### DISCUSSION TITLE
Pydna Hackathon - Thursday 3rd of October

### DISCUSSION THREAD

**Discussion by manulera** [pydna-group/pydna#258] [General]
## Joining information

* We will have a Teams / Zoom meeting running during the hackathon. All the details in the document below, since joining links and all may change.
* If you are planning to join us, check the info and indicate when you would be able to join!

https://docs.google.com/document/d/1NLJu9zzzucDaIkoiHchsPJRawbD25umzM_TW2zqihXY/edit?usp=sharing

## General info

We discussed to do a Hackathon this month. The idea is to commit a chunk of time (at least 1-2 hours) to work on pydna issues. To make the most of the day, we should agree in advance about what we are going to work on during the day.

1. Date: Thursday 3rd of October
1. Mention things that could be worked on by mentioning them in this discussion. Also include an estimate of the time they would take to implement a first version. The idea is that at least a first version of the feature can be completed within the day, so should not be a massive implementation.
2. Create an assign issues tagged with the label [Hackathon-2024](https://github.com/BjornFJohansson/pydna/labels/Hackathon-2024).

Once they are added to the discussion, I will include them in a list in this post, so that we can keep track on what's on the menu for the day. In the meantime, check the [issues tagged with hackathon](https://github.com/BjornFJohansson/pydna/issues?q=is%3Aissue+is%3Aopen+label%3AHackathon-2024).

On the day, we can keep a zoom room open where people can jump in, introduce themselves, ask questions, etc.

## Documentation to get started

We have recently put together some documentation as jupyter notebooks that can be found [here](https://github.com/BjornFJohansson/pydna/tree/dev_bjorn/docs/notebooks). If you have some feedback on it, you can add it to this issue: https://github.com/BjornFJohansson/pydna/issues/260

**Comment by manulera** [comment]
## improve assembly fragments

@hiyama341 you mentioned an improvement to assembly_fragments [here](https://github.com/hiyama341/test_assembly_fragments_pydna/blob/main/notebooks/00-LL_test_assembly_fragments.ipynb). If this is something you would be interested in doing, you can create an issue and self-assign.

**Comment by manulera** [comment]
## typing

There is an existing issue https://github.com/BjornFJohansson/pydna/issues/176.

I think this is a good one because it's a bit boring so you would not do it on a normal day, but it's quite useful and everyone can chip in no matter how much time they have.

I will probably work on this myself during the hackathon

**Comment by manulera** [comment]
## Contributing to pydna hackathon?

Hi, sorry for the spammy comment but I am tagging everyone who has contributed / created an issue in case they are interested in participating in the hackathon directly, or in proposing issues to work on for the participants.

@mountainpenguin
@JamesBagley
@kcorreia
@JacobHayes
@bruno2git
@benweinberg89
@ngamarra
@santirdnd
@glebkuznetsov
@photocyte
@marcus-r-kelly
@LuisaFJimenez
@timovwb
@JeffXiePL
@louisabraham
@andrewdavis3
@cbrueffer
@FinweNoldoran
@jefe-jeff
@shenker
@manulera
@hgbrian
@amijalis
@jlerman44
@zslastman
@grassoste
@jxu10
@osthomas
@mohamedmoez1983
@hsiaut
@TCoulth
@ghost
@AubinF
@harijay
@dependabot[bot]
@nick-chavez23
@surgebiswas
@hiyama341
@jan-glx
@willsharpless
@klprint
@hristog
@dgruano
@picousse
@BjornFJohansson
@codacy-badger
@ITS-Account
@mcrone

**Reply by manulera** [reply]
Hello @louisabraham @harijay @mdbetterton we have set the 3rd of October as the date, according to availability from when2meet.

I will probably join from 8:00 to 18:00 UK time, and then re-join at 20:00 until bedtime if someone from a different timezone is still around. Feel free to write what times work for you at some point if you want to join on Zoom!

**Reply by manulera** [reply]
@louisabraham @harijay @mdbetterton @ExotoxinsLab I have added some joining information for the day, feel free to ask further questions!

**Comment by manulera** [comment]
## guided struggle

The hackathon can also be an occasion for people to try the library for the first time knowing that we are going to be around to answer questions, so if there is interest from people, we could also propose to do that. They can use the day to try the library on their own and jump on the call if they get stuck with something.

If you are interested on participating on this, please reply to this comment!

**Reply by mdbetterton** [reply]
I would be interested in trying the library for the first time, if that winds up being part of the event.

**Reply by manulera** [reply]
Hi @mdbetterton yes, definitely!

**Reply by ExotoxinsLab** [reply]
I would like to come! :smile:

**Comment by harijay** [comment]
I am interested in participating .

Thanks for sending out the request

Hari

**Reply by manulera** [reply]
Thanks Hari, very appreciated, feel free to give your availability in the when2meet link in the top comment and to share some things you would like to see fixed / improved. You can keep an eye on the [issues tagged with hackathon](https://github.com/BjornFJohansson/pydna/issues?q=is%3Aissue+is%3Aopen+label%3AHackathon-2024) and assign to either of them.

**Comment by louisabraham** [comment]
I would love to implement some algorithms or make existing implementations faster. Any ideas?

**Reply by manulera** [reply]
@BjornFJohansson any ideas for this? I have not thought much about performance

**Reply by BjornFJohansson** [reply]
Hi, Louis
I never did a formal profiling of the code. The test suite always spent more time finding shared substrings, now that is much faster thanks to pydivsufsort. Some of the __init__ for Dseq and Dseqrecord are slow since they are quite flexible (though there are quicker alternative versions). Perhaps the graph algorithms for finding paths in the homologous recombination function could be an interesting target. The  current implementation is pure python via NetworkX, bu there are several faster libraries.

**Comment by dgruano** [comment]
I have some scripts that I used to design synonimous gene sequences with (1) mutated PAMs and (2) the addition or removal of restriction sites. I found them super helpful. If you think these could be integrated within `pydna`, I'm happy to give it a shot!

**Comment by harijay** [comment]
Hi All
I was really hoping to join the hackathon  , but am down with the flu.
Sincere apologies for the last minute notice.

Here’s hoping for more of these in the future

Hari
